docs/platform/connector-development/connector-builder-ui/error-handling.md

```diff
@@ -67,57 +130,187 @@ The [Recurly API](https://recurly.com/developers/api/v2021-02-25/index.html#sect
+
+**Examples:**
+- `"{{ 'Too much requests' in response }}"` (check if text exists in response)
+- `"{{ response.code == 300 }}"` (check specific field value)
+- `"{{ 'error_code' in response and response['error_code'] == 'ComplexityException' }}"` (complex condition)
 
 ##### Example
 
-For the Zoom API, the response text body can include a special non-error status codes under the `code` field. An example response text body would look like `{"code": 300}`. The "If error message matches" condition is too broad because there could be record data containing the text "300". Instead, for a response filter defining "and predicate is fulfilled" as `{{ response.code == 300 }}`, during a sync, the predicate expression will be evaluated to true and the connector proceeds based on the "Then execute action".
+For the Zoom API, the response text body can include a special non-error status codes under the `code` field. An example response text body would look like `{"code": 300}`. The "Error message contains" condition is too broad because there could be record data containing the text "300". Instead, for a response filter defining the predicate as `{{ response.code == 300 }}`, during a sync, the predicate expression will be evaluated to true and the connector proceeds based on the configured action.
+
+#### HTTP status codes
 
-#### and HTTP codes match
+A response filter can specify a set of numeric HTTP status codes to match against. When receiving an API response, the connector will check if the status code of the response is in the provided set of HTTP status codes.
 
-A response filter can specify for the "and HTTP codes match" field a set of numeric HTTP status codes (ex. 200, 404, 500). When receiving an API response, the connector will check to see if the status code of the response is in the provided set of HTTP status codes.
+**Configuration:**
+- **HTTP codes**: An array of HTTP status codes to match against
+
+**Examples:**
+- `[420, 429]` (match rate limiting codes)
+- `[500]` (match server errors)
+- `[403, 404]` (match specific client errors)
 
 ##### Example
 
-The Pocket API emits API responses for rate limiting errors using a 403 error status code. The default error handler interprets 403 errors as non-retryable and will fail the sync when they are encountered. The connector can configures a response filter field "and HTTP status codes" that contains 403 within the set. When a 403 error response from the API is encountered, the connector proceeds based on the "Then execute action"
+The Pocket API emits API responses for rate limiting errors using a 403 error status code. The default error handler interprets 403 errors as non-retryable and will fail the sync when they are encountered. The connector can configure a response filter with HTTP status codes that contains 403. When a 403 error response from the API is encountered, the connector proceeds based on the configured action.
+
+### Response Actions
```

Comment:
We may also want to put the Action and Failure types sections above the other ones, to match the order they are shown in the UI